org.eclipse.m2e.pde.ui/src/org/eclipse/m2e/pde/ui/target/editor/internal/TargetDependencyModel.java

```diff
@@ -0,0 +1,339 @@
+			int index = oldTargetDependencies.indexOf(oldCurrentSelection.get(0));
+
+			// Is the index still valid? If not, select the last element instead...
+			if (index >= newTargetDependencies.size()) {
+				index = newTargetDependencies.size() - 1;
+			}
+
+			MavenTargetDependency itemToSelect = newTargetDependencies.get(index);
+			newCurrentSelection = Collections.singletonList(itemToSelect);
+		}
+
+		history.modelChange(newTargetDependencies, newCurrentSelection);
+	}
+
+	/**
+	 * Updates the versions of all given artifacts to the latest one available on
+	 * Maven Central (or any additional repository specified in the target
+	 * location). If all dependencies are already up-to-date, nothing is done. An
+	 * error is logged if {@link canUpdate()} returns {@code false}.<br>
+	 * This operation is undoable.
+	 * 
+	 * @see #canUpdate()
+	 */
+	public void update() {
+		if (!canUpdate()) {
+			LOGGER.error("The update operation is currently not possible! Ignore...");
+			return;
+		}
+
+		List<MavenTargetDependency> oldCurrentSelection = getCurrentSelection();
+		List<MavenTargetDependency> oldTargetDependencies = getTargetDependencies();
+
+		List<MavenTargetDependency> newCurrentSelection = new ArrayList<>();
+		List<MavenTargetDependency> newTargetDependencies = new ArrayList<>(oldTargetDependencies);
+
+		try {
+			int updated = 0;
+
+			for (MavenTargetDependency dependency : oldCurrentSelection) {
+				int index = oldTargetDependencies.indexOf(dependency);
+
+				MavenTargetDependency newDependency = targetLocation.update(dependency, null);
+
+				if (!dependency.matches(newDependency)) {
+					updated++;
+				}
+
+				newTargetDependencies.set(index, newDependency);
+				newCurrentSelection.add(newDependency);
+			}
+
+			// An "empty" update should not show up on the command stack...
+			if (updated > 0) {
+				history.modelChange(newTargetDependencies, newCurrentSelection);
+			}
+		} catch (CoreException e) {
+			LOGGER.error(e.getMessage(), e);
+		}
+	}
+
+	/**
+	 * Checks whether more than one element is selected.
+	 * 
+	 * @return {@code true}, if the update remove can be performed.
+	 * @see #remove()
+	 */
+	public boolean canRemove() {
+		return !currentSelection.isEmpty();
+	}
+
+	/**
+	 * Checks whether all selected element contain a valid GAV.
+	 * 
+	 * @return {@code true}, if the update operation can be performed.
+	 * @see #update()
+	 */
+	public boolean canUpdate() {
+		if (targetLocation == null) {
+			return false;
+		}
+
+		for (MavenTargetDependency dependency : currentSelection) {
+			if (dependency.getGroupId() == null || dependency.getGroupId().isBlank()) {
+				return false;
+			}
+			if (dependency.getArtifactId() == null || dependency.getArtifactId().isBlank()) {
+				return false;
+			}
+			if (dependency.getVersion() == null || dependency.getVersion().isBlank()) {
+				return false;
+			}
+		}
+
+		return !currentSelection.isEmpty();
```

Comment:
```suggestion
		Stream<String> allFields = currentSelection.stream().mapMulti((dependency, downStream) -> {
			downStream.accept(dependency.getGroupId());
			downStream.accept(dependency.getArtifactId());
			downStream.accept(dependency.getVersion());
		});
		return allFields.noneMatch(StringUtils::isBlank) && !currentSelection.isEmpty();
```

Reply:
That looks a lot cleaner. Thanks for the feedback!